openpgp/read.go

```diff
@@ -163,7 +163,10 @@ FindKey:
 			}
 			if !pk.key.PrivateKey.Encrypted {
 				if len(pk.encryptedKey.Key) == 0 {
-					pk.encryptedKey.Decrypt(pk.key.PrivateKey, config)
+					errDec := pk.encryptedKey.Decrypt(pk.key.PrivateKey, config)
+					if errDec != nil {
+						return nil, errDec
+					}
 				}
 				if len(pk.encryptedKey.Key) == 0 {
 					continue
```

Comment:
This error is intentionally ignored, I think, because we could have multiple session key packets, only one of which may be valid. Hence the check for 
```go
				if len(pk.encryptedKey.Key) == 0 {
					continue
				}
```
But - probably it would be good to make this more explicit and do something like:
```suggestion
					errDec := pk.encryptedKey.Decrypt(pk.key.PrivateKey, config)
					if errDec != nil {
						continue
					}
```
Especially because that function seems to only check the checksum after writing the decrypted key:
```go
	e.Key = b[1 : len(b)-2]
	expectedChecksum := uint16(b[len(b)-2])<<8 | uint16(b[len(b)-1])
	checksum := checksumKeyMaterial(e.Key)
	if checksum != expectedChecksum {
		return errors.StructuralError("EncryptedKey checksum incorrect")
	}
```
so actually this is a good catch :+1:

Reply:
Right, I agree; is this not tested, actually? Because if we had a test for multiple key packets, my `return nil, errDec` would have broken that.

Reply:
Yeah, apparently not 😕